src/commonMain/kotlin/exchange.dydx.abacus/state/v2/manager/StateManagerAdaptorV2.kt

```diff
@@ -589,4 +633,37 @@ internal class StateManagerAdaptorV2(
             )
         }
     }
+
+    private fun didSetGeo(geo: String?) {
+        val state = stateMachine.state
+        stateMachine.state = PerpetualState(
```

Comment:
kotlin tip: `data class` has a built-in `copy()` method:
`stateMachine.state = stateMachine.state.copy(compliance = ...)`